torch/testing/__init__.py

```diff
@@ -140,7 +157,7 @@ def _compare_tensors_internal(a: torch.Tensor, b: torch.Tensor, *, rtol, atol, e
 
     # Masks close values
     # NOTE: this avoids (inf - inf) oddities when computing the difference
-    close = torch.isclose(a_flat, b_flat, rtol, atol, equal_nan)
+    close = torch.isclose(a_flat, b_flat, rtol, atol, bool(equal_nan))
```

Comment:
Same change to `bool(equal_nan)` here (or do them both at once in a local variable)